astropy/io/fits/hdu/image.py

```diff
@@ -245,6 +245,14 @@ def data(self):
 
         return data
 
+    @lazyproperty
+    def wcs(self):
+        """
+        Returns the WCS of the images
+        """
+
+        return wcs.WCS(self.header)
```

Comment:
This will return an incomplete WCS object if part of the distortion is in separate extensions. To initialize the WCS correctly pass the entire FITS HDUList object.